src/briefcase/console.py

```diff
@@ -55,7 +55,9 @@ class Printer:
     """Interface for printing and managing output to the console and/or log."""
 
     # Console to manage console output.
-    console = RichConsole(highlighter=RichConsoleHighlighter(), emoji=False)
+    console = RichConsole(
+        highlighter=RichConsoleHighlighter(), emoji=False, soft_wrap=True
+    )
```

Comment:
I had to add `soft_wrap` to prevent Rich from wrapping everything at 80 columns in an MSYS2 terminal regardless of the window size. I'm not sure when this started happening: going back to commit 8df2d11 didn't make any difference. It might even have been caused by yesterday's Windows update. 

@rmartin16: Have you ever seen a problem like this? And can you confirm that this change should be harmless?

Reply:
I haven't seen this....but it (obviously) implies that Rich cannot determine the size of the terminal....although, if it cannot determine that, it probably knows next to nothing about the terminal...

I can't get `briefcase` installed using MSYS2; `psutils` says `cygwin` isn't supported.

Out of curiosity, what does `python -m rich.diagnose` print?

As far as `soft_wrap=True`, it won't break anything...but it's unfortunate since its prior setting would prevent words bring broken over line breaks and increased readability IMO.

If this change stays, you can also remove `Printer.console.soft_wrap = True` from `tests/conftest.py`.

Reply:
ahh...ok...learning new things today. MSYS2 can run windows binaries...seems to be a main selling point...anyway

I can replicate it using Python installed via Windows. This is probably coming from Rich setting `legacy_windows=True`.

<details>
<summary>python.exe -m rich.diagnose</summary>

```
$ ./venv-win/Scripts/python.exe -m rich.diagnose
+---------------------- <class 'rich.console.Console'> -----------------------+
| A high level console interface.                                             |
|                                                                             |
| +-------------------------------------------------------------------------+ |
| | <console width=79 None>                                                 | |
| +-------------------------------------------------------------------------+ |
|                                                                             |
|     color_system = None                                                     |
|         encoding = 'cp1252'                                                 |
|             file = <_io.TextIOWrapper name='<stdout>' mode='w'              |
|                    encoding='cp1252'>                                       |
|           height = 25                                                       |
|    is_alt_screen = False                                                    |
| is_dumb_terminal = False                                                    |
|   is_interactive = False                                                    |
|       is_jupyter = False                                                    |
|      is_terminal = False                                                    |
|   legacy_windows = True                                                     |
|         no_color = False                                                    |
|          options = ConsoleOptions(                                          |
|                        size=ConsoleDimensions(width=79, height=25),         |
|                        legacy_windows=True,                                 |
|                        min_width=1,                                         |
|                        max_width=79,                                        |
|                        is_terminal=False,                                   |
|                        encoding='cp1252',                                   |
|                        max_height=25,                                       |
|                        justify=None,                                        |
|                        overflow=None,                                       |
|                        no_wrap=False,                                       |
|                        highlight=None,                                      |
|                        markup=None,                                         |
|                        height=None                                          |
|                    )                                                        |
|            quiet = False                                                    |
|           record = False                                                    |
|         safe_box = True                                                     |
|             size = ConsoleDimensions(width=79, height=25)                   |
|        soft_wrap = False                                                    |
|           stderr = False                                                    |
|            style = None                                                     |
|         tab_size = 8                                                        |
|            width = 79                                                       |
+-----------------------------------------------------------------------------+
+--- <class 'rich._windows.WindowsConsoleFeatures'> ----+
| Windows features available.                           |
|                                                       |
| +---------------------------------------------------+ |
| | WindowsConsoleFeatures(vt=False, truecolor=False) | |
| +---------------------------------------------------+ |
|                                                       |
| truecolor = False                                     |
|        vt = False                                     |
+-------------------------------------------------------+
+------ Environment Variables -------+
| {                                  |
|     'TERM': 'xterm',               |
|     'COLORTERM': None,             |
|     'CLICOLOR': None,              |
|     'NO_COLOR': None,              |
|     'TERM_PROGRAM': 'mintty',      |
|     'COLUMNS': None,               |
|     'LINES': None,                 |
|     'JUPYTER_COLUMNS': None,       |
|     'JUPYTER_LINES': None,         |
|     'JPY_PARENT_PID': None,        |
|     'VSCODE_VERBOSE_LOGGING': None |
| }                                  |
+------------------------------------+
platform="Windows"
```
</details>

Reply:
with that in mind, maybe `soft_wrap` should be set with something like ` shutil.get_terminal_size((1, 1)) == (1, 1)`

Reply:
> MSYS2 can run windows binaries...seems to be a main selling point.

Yes, I'm using the standard Windows Python from python.org, I just use MSYS2 to get a Unix-like shell.

`python -m rich.diagnose` gives similar output to you (width = 79, height = 25), and shows the environment variables `COLUMNS` and `LINES` both as None. Both of those variables are actually set to the correct values in Bash, but not exported, so they're not available to Python. If I do export them, then `rich.diagnose` displays the correct width and height.

Even if they're not exported, `os.get_terminal_size()` returns the correct result anyway:
```
os.terminal_size(columns=105, lines=61)
```
So I don't know where that's coming from or why Rich was failing to use it.

> it won't break anything...but it's unfortunate since its prior setting would prevent words bring broken over line breaks and increased readability IMO

Word wrapping might be a bit more readable, but it has the disadvantage that when someone copies the text from the terminal into an issue report, it'll include line breaks that weren't there in the original. For example, if a phrase gets split up by a line break, it might prevent it from being found in a search for that phrase, either using GitHub's search  function or a web browser's "search within page" command.

Reply:
I agree with @mhsmith; while word wrapping is nice if you're dealing with text, we're dealing with content that is frequently mixed with computer-generated content or URLs, and the exact spacing and newlines matter in that context. 

Beyond that - my reading and testing seems to suggest `soft_wrap` is effectively a no-op in this case, so I'm OK with adding it if it cleans up a problem case on MSYS2.